src/main/java/wenjigglybot/Parser.java

```diff
@@ -0,0 +1,138 @@
+package wenjigglybot;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
+public class Parser {
+
+    /**
+     * Processes the input string representing an event task.
+     * Extracts the event description, start time, and end time.
+     *
+     * @param task The input string containing the event details.
+     * @return A string array containing the event description, start time, and end time.
+     * @throws EventException If the input string does not contain the correct format.
+     */
+    public static String[] processEventTask(String task) throws EventException {
+        String[] fromParts = task.split("/from");
+        if (fromParts.length != 2) {
+            throw new EventException();
+        }
+        String[] toParts = fromParts[1].split("/to");
+        if (toParts.length != 2) {
+            throw new EventException();
+        }
+        // Extract the event description, start time, and end time
+        String event = fromParts[0].replaceFirst("event", "").trim();
+        String startTime = toParts[0].trim();
+        String endTime = toParts[1].trim();
+        return new String[]{event, startTime, endTime};
+    }
+
+    /**
+     * Processes the input string representing a deadline task.
+     * Extracts the task name and deadline.
+     *
+     * @param task The input string containing the deadline details.
+     * @return A string array containing the task name and deadline.
+     * @throws DeadlineException If the input string does not contain the correct format.
+     */
+    public static String[] processDeadlineTask(String task) throws DeadlineException {
+        // Remove deadline tag
+        String taskNameAndDeadline = task.replaceFirst("deadline", "").trim();
+
+        // Split the title and deadline
+        String[] parts = taskNameAndDeadline.split("/by");
+        if (parts.length != 2) {
+            throw new DeadlineException();
+        }
+        return parts;
+    }
+
+    /**
+     * Parses the input command string to determine the corresponding command.
+     *
+     * @param command The input command string.
+     * @return The corresponding {@link Command} enum value.
+     * @throws InvalidCommandException If the command is not recognized.
+     */
+    public static Command parseCommand(String command) throws InvalidCommandException {
+        for (Command cmd : Command.values()) {
+            if (command.startsWith(cmd.name().toLowerCase())) {
+                return cmd;
+            }
+        }
+        throw new InvalidCommandException();
+    }
+
+    /**
+     * Parses a string representation of a task and returns the corresponding Task object.
+     *
+     * @param line The string representing the task.
+     * @return The corresponding {@link Task} object, or null if the task could not be parsed.
+     */
+    public static Task parseTask(String line) {
+        // Example format: 1. [T][X] task description
+        String[] parts = line.split(" ", 3);
```

Comment:
Maybe change this name of parts to commandParts for more clarity?

Reply:
👍